trees/isolation.go

```diff
@@ -0,0 +1,234 @@
+package trees
+
+import (
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/sjwhitworth/golearn/base"
+)
+
+type IsolationForest struct {
+	nTrees   int
+	maxDepth int
+	subSpace int
+	trees    []regressorNode
+}
+
+// Select A random feature for splitting from the data.
+func selectFeature(data [][]float64) int64 {
+	rand.Seed(time.Now().UnixNano())
```

Comment:
Re-seeding the random number generator will make things harder to reproduce exactly. It may be best to leave this to the program using the library.

Reply:
Hi,

So the gaussian_outliers.csv was generated by me. It is an example of a standard type of dataset used for outlier detection. It essentially has a lot of data points from gaussian data (which are normal), and then we have 10 outliers I added.

I think the idea behind an isolation forest is that everything is as random as possible. Only when this condition is met, outliers are filtered out early. That's why I changed  the rand.seed, because otherwise the seed is fixed and random numbers are totally predictable. 

So then I'll remove the lines that change the seed and add another commit?